### PR TITLE
#317: Checkbox component

### PR DESCRIPTION
**Pull request by BeeMargarida**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-design/issues/317|
| Dependencies | |
| Decisions | - Checkbox similar to ripe-components-vue <br> - Support for active transition, like components-vue (new touchable props) <br> - Svgs equal to the ones in components-vue <br> Tested this yesterday in iOS and it seemed fine as well (similar to android) <br> <br> NOTE: A lesson learned while doing this, the normal reload does not catch the changes made to SVGs. `yarn run clear` was needed (after a day of smashing my head, wondering why the svg did not resize, even with viewBox). |
| Animated GIF | checkbox|

### CODE REVIEW

File: react/components/atoms/checkbox/checkbox.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+import { Platform, StyleSheet, Text, View, ViewPropTypes } from "react-native";
+import PropTypes from "prop-types";
+import { mix } from "yonius";
+
+import { IdentifiableMixin, baseStyles } from "../../../util";
+
+import { Icon } from "../icon";
+import { Touchable } from "../touchable";
+
+export class Checkbox extends mix(PureComponent).with(IdentifiableMixin) {
```
Comment (beneditomauro) [inline react/components/atoms/checkbox/checkbox.js:11]: Shouldn't you id this component with the `IdentifiableMixin` ?

**Review by beneditomauro — APPROVED** [review]

File: react/components/atoms/checkbox/checkbox.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        );
+    };
+
+    _icon = () => {
+        if (this.props.disabled) return `${this.props.icon}-gray`;
+        if (this.state.active) return `${this.props.icon}-blue`;
+        return this.props.icon;
+    };
+
+    _style = () => {
```
Comment (gcandal) [inline react/components/atoms/checkbox/checkbox.js:75]: We're reserving `_style` for the root component, so call this something else (e.g. styleBox)

File: react/components/atoms/checkbox/checkbox.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        if (this.state.active) base.push(styles.checkboxBoxActive);
+        if (this.props.variant === "error") base.push(styles.checkboxBoxError);
+        if (this.props.disabled) base.push(styles.checkboxBoxDisabled);
+
+        base.push(this.props.style);
+        return base;
+    };
+
+    _textStyle = () => {
+        const base = [styles.checkboxLabel];
```
Comment (gcandal) [inline react/components/atoms/checkbox/checkbox.js:93]: Keep the naming coherent: it's either "text" or "label"

File: react/components/atoms/checkbox/checkbox.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        const base = [
+            styles.checkboxBox,
+            {
+                width: this.props.size,
+                height: this.props.size
+            }
+        ];
+        if (this.state.checkedData) base.push(styles.checkboxBoxChecked);
+        if (this.state.active) base.push(styles.checkboxBoxActive);
+        if (this.props.variant === "error") base.push(styles.checkboxBoxError);
```
Comment (gcandal) [inline react/components/atoms/checkbox/checkbox.js:85]: Use a similar approach to other components: https://github.com/ripe-tech/ripe-components-react-native/blob/c8934fcf990cc6fbb991988cbe5a0173bd78ee0a/react/components/atoms/button-tab-text/button-tab-text.js#L42

**Comment by gcandal** [comment]
Check block vs inline-block (as we talked)

**Review by gcandal — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**